instrumentation/netty/netty-4.1/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/netty/v4_1/Netty41ClientSslTest.java

```diff
@@ -0,0 +1,260 @@
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpClientTestServer;
+import io.opentelemetry.sdk.testing.assertj.AttributeAssertion;
+import io.opentelemetry.sdk.trace.data.StatusData;
+import io.opentelemetry.semconv.ExceptionAttributes;
+import io.opentelemetry.semconv.NetworkAttributes;
+import io.opentelemetry.semconv.ServerAttributes;
+import java.net.URI;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLHandshakeException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class Netty41ClientSslTest {
+
+  @RegisterExtension
+  static InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  private HttpClientTestServer server;
+
+  private EventLoopGroup eventLoopGroup;
+
+  @SuppressWarnings("UnusedMethod")
+  private static Bootstrap createBootstrap(
+      EventLoopGroup eventLoopGroup, List<String> enabledProtocols) {
+    Bootstrap bootstrap = new Bootstrap();
+    bootstrap
+        .group(eventLoopGroup)
+        .channel(NioSocketChannel.class)
+        .handler(
+            new ChannelInitializer<SocketChannel>() {
+              @Override
+              protected void initChannel(SocketChannel socketChannel) throws Exception {
+                ChannelPipeline pipeline = socketChannel.pipeline();
+
+                SslContext sslContext = SslContextBuilder.forClient().build();
+                SSLEngine sslEngine = sslContext.newEngine(socketChannel.alloc());
+                if (enabledProtocols != null) {
+                  sslEngine.setEnabledProtocols(enabledProtocols.toArray(new String[0]));
+                }
+                pipeline.addLast(new SslHandler(sslEngine));
+
+                pipeline.addLast(new HttpClientCodec());
+              }
+            });
+    return bootstrap;
+  }
+
+  @BeforeEach
+  void setUp() {
+    server = new HttpClientTestServer(testing.getOpenTelemetry());
+    server.start();
+    eventLoopGroup = new NioEventLoopGroup();
+    System.setProperty("otel.instrumentation.netty.connection-telemetry.enabled", "true");
+    System.setProperty("otel.instrumentation.netty.ssl-telemetry.enabled", "true");
+  }
+
+  @AfterEach
+  void tearDown() throws ExecutionException, InterruptedException, TimeoutException {
+    server.stop().get(10, TimeUnit.SECONDS);
+    eventLoopGroup.shutdownGracefully();
+    System.clearProperty("otel.instrumentation.netty.connection-telemetry.enabled");
+    System.clearProperty("otel.instrumentation.netty.ssl-telemetry.enabled");
+  }
+
```

Comment:
groovy version does these only once for class

Reply:
done